### ISSUE TITLE
Unstable performance

### ISSUE THREAD

**Issue by tjulyz** [dvornikita/blitznet#32]
Hi! Thanks for your kind share! I have met some problems when I used your code to train the models to reproduce the performance as you list in your paper.
With the same configuration (batch size=32, step=40000 for Pascal voc2012), I trained the network for several times. Then I found that the values of mIOU and mAP are unstable. The amplitude of variations is about 1%. Have you ever met this problem? Moreover, I can’t trained a better model to get the detection performance as you list in Table 3 in your paper. My best is about 79.1 (80.0 in Table 3). Can you help me for a better performance reproduction? Thanks a lot !

**Comment by dvornikita** [comment]
The performance may be unstable indeed. This is a common problem of object detectors that have to solve a different problem in training and test time. NMS is not robust to small changes in boxes scores either. 
Why do you mean by «step=4000 »?

**Comment by dvornikita** [comment]
--step is a parameter used in testing (how often you evaluate checkpoints). See config.py for details.
What you meant is probably --lr_decay (see code snippets on how to train the model). The learning rate is decreased twice during the training. For details check the original paper.

**Comment by tjulyz** [comment]
Thanks for your reply! I set the configurations for voc2012 as (lr_decay 25000 35000) for training as that in your paper.  Have you evaluated all/some checkpoints  (e.g. --step=2) when you reported your performance in your paper? If so, whether the result is calculated by average or max?

**Comment by dvornikita** [comment]
I see what you mean now. 
We simply took the last checkpoint for each model, usually, its performance was the best or among the best (in ~0.3% range). If you train the model several times and none of the results if close enough to the reported ones it may indicate a problem, especially if it happens consistently in other settings as well. Otherwise, it's possible to observe performance fluctuating around the reported values.

**Comment by tjulyz** [comment]
Thanks a lot! I am trying to find the training problem.
Have you used warmup_lr in your training? I find the warmup_lr in the config is set as 1e-5, but the warmup_step is set to 0 as default.  
If you use warmup_lr, how many steps have you set to train the network?
Thanks a lot!

**Comment by dvornikita** [comment]
@tjulyz warmup was meant to help with some other experiments. In the end, we didn’t use it in the experiments reported (that’s why it’s set to 0)
What do you mean by “training problem”?

**Comment by tjulyz** [comment]
Thanks for your kind reply! 
Because I haven't reproduced the performance, I am trying to find out if there are some problems when I training the network.
I will close this issues soon. If I find the answer, I will discuss it again.
Thanks a lot!
